### PR TITLE
Satori implement new functions  pull request

### PR DESCRIPTION
**Pull request by miler012**
hi. we had implement new functions as you say.
pls have a check

### CODE REVIEW

File: adapters/satori/src/sdk/subgraphDetailsV2.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+
+export const getUserTVLByBlock = async (
+    blockNumber: number,
+    timestamp: number,
+):Promise<OutputDataSchemaRow[]> =>  {
+    let subgraphUrl = SUBGRAPH_URL;
+    let blockQuery = blockNumber !== 0 ? `block: {number: ${blockNumber}}` : ``;
+    let skip = 0;
+    let result: OutputDataSchemaRow[] = [];
```
Comment (nitish-91) [inline adapters/satori/src/sdk/subgraphDetailsV2.ts:20]: @miler012 do you need to put this in a while loop to make sure all records related to a block number are fetched.

File: adapters/satori/src/sdk/subgraphDetailsV2.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+          }
+          `;
+        
+        let response = await fetch(subgraphUrl, {
+            method: "POST",
+            body: JSON.stringify({ query }),
+            headers: { "Content-Type": "application/json" },
+        });
+        let data = await response.json();
+        console.log(data);
```
Comment (nitish-91) [inline adapters/satori/src/sdk/subgraphDetailsV2.ts:36]: @miler012 please dont log the whole result

**Comment by miler012** [comment]
hi. we did the upgrade,pls check

**Comment by 0xroll** [comment]
hi @miler012 ,

is this for both perps + the v2 LP data you guys have?

not sure if the v2 lp data is included in here, if not please add it in as well. you can refer to sushi's PR for v2 AMM LP data

https://github.com/delta-hq/l2-lxp-liquidity-reward/pull/37/files

can i have the subgraph repo link as well?

**Comment by miler012** [comment]
hi. just  perps

**Comment by miler012** [comment]
and this is the subgragh link  https://api.goldsky.com/api/public/project_cltzggx5i40gi01p3ezjv94bc/subgraphs/satori-linea/v0.0.1/gn

**Pull request merged**